File: library/src/main/java/com/getstream/sdk/chat/model/StreamNotification.java
package com.getstream.sdk.chat.model;

import android.app.PendingIntent;

import androidx.annotation.Nullable;

import com.google.firebase.messaging.RemoteMessage;

public class StreamNotification {

    private int notificationId;
    private String channelName;
    private String messageText;
    private PendingIntent pendingIntent;
    private RemoteMessage remoteMessage;
    private Event event;

    public StreamNotification(int notificationId, @Nullable RemoteMessage remoteMessage, @Nullable Event event) {
        this.notificationId = notificationId;
        this.remoteMessage = remoteMessage;
        this.event = event;
    }

    public int getNotificationId() {
        return notificationId;
    }

    public String getChannelName() {
        return channelName;
    }

    public String getMessageText() {
        return messageText;
    }

    public PendingIntent getPendingIntent() {
        return pendingIntent;
    }

    public RemoteMessage getRemoteMessage() {
        return remoteMessage;
    }

    public Event getEvent() { return event; }

    public void setChannelName(String channelName) {
        this.channelName = channelName;
    }

    public void setMessageText(String messageText) {
        this.messageText = messageText;
    }

    public void setPendingIntent(PendingIntent pendingIntent) {
        this.pendingIntent = pendingIntent;
    }
}

File: app/src/main/java/io/getstream/chat/android/core/poc/library/models/StreamNotification.kt
package io.getstream.chat.android.core.poc.library.models

import android.app.PendingIntent
import com.google.firebase.messaging.RemoteMessage
import io.getstream.chat.android.core.poc.library.events.ChatEvent

class StreamNotification constructor(
    val notificationId: Int = 0,
    val remoteMessage: RemoteMessage? = null,
    val event: ChatEvent? = null
) {

    var channelName: String? = null
    var messageText: String? = null
    var pendingIntent: PendingIntent? = null
}